src/components/payroll/AdvancedFiltersDialog.js

```diff
@@ -0,0 +1,284 @@
+/* eslint-disable no-unused-vars */
+/* eslint-disable react/jsx-no-useless-fragment */
+/* eslint-disable no-prototype-builtins */
+import React, { useEffect, useState } from 'react';
+import { injectIntl } from 'react-intl';
+import Button from '@material-ui/core/Button';
+import Dialog from '@material-ui/core/Dialog';
+import DialogActions from '@material-ui/core/DialogActions';
+import DialogContent from '@material-ui/core/DialogContent';
+import DialogTitle from '@material-ui/core/DialogTitle';
+import {
+  decodeId,
+  formatMessage,
+  fetchCustomFilter,
+} from '@openimis/fe-core';
+import { withTheme, withStyles } from '@material-ui/core/styles';
+import { connect } from 'react-redux';
+import { bindActionCreators } from 'redux';
+import AddCircle from '@material-ui/icons/Add';
+import AdvancedFiltersRowValue from './AdvancedFiltersRowValue';
+import { BENEFIT_PLAN, CLEARED_STATE_FILTER } from '../../constants';
+import { isBase64Encoded, isEmptyObject } from '../../utils/advanced-filters-utils';
+
+const styles = (theme) => ({
+  item: theme.paper.item,
+});
+
+function AdvancedFiltersDialog({
```

Comment:
The logic for applying filter is already on core. But there is a huge difference in using this in terms of having value that is saved on particular object in json_ext field. We can think about this later when we have time. At this stage it could take a lot of time to move this to core module and make more generic.